libraries/testbot/dialogs/mainDialog.js

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { TimexProperty } = require('@microsoft/recognizers-text-data-types-timex-expression');
+const { ComponentDialog, DialogSet, DialogTurnStatus, TextPrompt, WaterfallDialog } = require('botbuilder-dialogs');
+const { BookingDialog } = require('./bookingDialog');
+
+const MAIN_WATERFALL_DIALOG = 'mainWaterfallDialog';
+const BOOKING_DIALOG = 'bookingDialog';
+
+class MainDialog extends ComponentDialog {
```

Comment:
We refactored MainDialog in the corebot-updates branch as follows:

- Updated constructor to take LUIS client and BookingDialog in the instructor so we can mock them in unit tests.
- Improved the flow so the dialog works with or without LUIS configured.
- Updated activities and prompts to set the speak and inputhint properties.
- Updated the finalStep to use ReplaceDialogAsync so the dialog is restarted at the end.
- Added warning message when the LUIS model recognizes the entity but doesn't resolve it to the Airport list.

The new C# version is here: https://github.com/microsoft/BotBuilder-Samples/blob/8dcaafc51a73f6663c7f53d07f3c9eaec0b68d5d/samples/csharp_dotnetcore/13.core-bot/Dialogs/MainDialog.cs#L23

Reply:
All of these items complete / ready for review with this commit.

Reply:
@jspruance, I did a second pass of mainDialog to match dotnet. We can review together on monday but now is working as expected.